<launch>
    <param name="/use_sim_time" value="true" />

    <arg name="world_file" default="$(find stage)/worlds/simple.world" />
    <arg name="laser_topic" default="base_scan" />

    <node pkg="stage_ros" type="stageros" name="stageros" required="true" args="$(arg world_file)" />

    <node name="cartographer_node" pkg="cartographer_ros"
          type="cartographer_node" args="
            -configuration_directory $(find stage_frontier_datagen)/config
            -configuration_basename cartographer.lua"
          output="screen">
        <remap from="scan" to="$(arg laser_topic)" />
    </node>

    <node name="cartographer_occupancy_grid_node" pkg="cartographer_ros" type="cartographer_occupancy_grid_node" />

    <node pkg="hector_driving_aid_markers" type="hector_driving_aid_markers_node" name="hector_driving_aid_markers" output="screen">
        <param name="left_side_y_outer" value="0.205"/>
        <param name="left_side_y_inner" value="0.11"/>
        <param name="right_side_y_outer" value="-0.205"/>
        <param name="right_side_y_inner" value="-0.11"/>
    </node>

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find stage_frontier_datagen)/config/hector_exploration.rviz">
        <remap from="base_scan" to="$(arg laser_topic)" />
    </node>
</launch>